src/DepthImageToLaserScanROS.cpp

```diff
@@ -31,61 +31,84 @@
  * Author: Chad Rockey
  */
 
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <sensor_msgs/msg/laser_scan.hpp>
+
+#include <functional>
+
 #include <depthimage_to_laserscan/DepthImageToLaserScanROS.h>
 
+#define ROS_ERROR printf
+
 using namespace depthimage_to_laserscan;
-  
-DepthImageToLaserScanROS::DepthImageToLaserScanROS(ros::NodeHandle& n, ros::NodeHandle& pnh):pnh_(pnh), it_(n), srv_(pnh) {
-  boost::mutex::scoped_lock lock(connect_mutex_);
-  
-  // Dynamic Reconfigure
-  dynamic_reconfigure::Server<depthimage_to_laserscan::DepthConfig>::CallbackType f;
-  f = boost::bind(&DepthImageToLaserScanROS::reconfigureCb, this, _1, _2);
-  srv_.setCallback(f);
-  
-  // Lazy subscription to depth image topic
-  pub_ = n.advertise<sensor_msgs::LaserScan>("scan", 10, boost::bind(&DepthImageToLaserScanROS::connectCb, this, _1), boost::bind(&DepthImageToLaserScanROS::disconnectCb, this, _1));
+
+DepthImageToLaserScanROS::DepthImageToLaserScanROS(rclcpp::node::Node::SharedPtr & node):node_(node) {
+  rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_sensor_data;
+
+  custom_qos_profile.history = RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+  custom_qos_profile.depth = 50;
+  custom_qos_profile.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+  custom_qos_profile.durability = RMW_QOS_POLICY_DURABILITY_VOLATILE;
+
+  cam_info_sub_ = node_->create_subscription<sensor_msgs::msg::CameraInfo>("depth_camera_info",
+      std::bind(
+        &DepthImageToLaserScanROS::infoCb, this,
+        std::placeholders::_1),
+      custom_qos_profile);
+
+  depth_image_sub_ =
+    node_->create_subscription<sensor_msgs::msg::Image>("depth",
+      std::bind(&DepthImageToLaserScanROS::depthCb, this, std::placeholders::_1),
+      custom_qos_profile);
+
+  scan_pub_ = node_->create_publisher<sensor_msgs::msg::LaserScan>("scan", custom_qos_profile);
+
+  parameter_service_ = std::make_shared<rclcpp::parameter_service::ParameterService>(node_);
+
+  float scan_time = 0.033;
+  node_->get_parameter("scan_time", scan_time);
```

Comment:
Actually, it doesn't end up being a one-liner since I still need to declare the variables.  Further, get_parameter_or() doesn't seem to have a templated version for float, so I need to change the types of all of the variables to double, and then I'd have to cast back to float for the call to dtl_->set*().  Overall, it doesn't seem to me that changing this to get_parameter_or() is a win.  Thoughts?